02_TrackWeb/src/background.ts

```
@@ -0,0 +1,269 @@
+  }
+}
+
+// Timer function for active time tracking
+function startActiveTimeTimer(tabId: number, normalizedUrl: string, domain: string) {
+  if (activeTabTimer !== null) {
+    clearInterval(activeTabTimer);
+  }
+  activeTabTimer = setInterval(() => {
+    // Only increment if the tab is still active and the window is focused
+    chrome.tabs.query({ active: true, currentWindow: true }, (tabs) => {
+      if (tabs && tabs[0] && tabs[0].id === tabId) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)